src/MoBi.Presentation/Tasks/CheckNameVisitor.cs

```diff
@@ -120,48 +120,81 @@ public void Visit(IFormula formula)
             }
 
             // possible alias Adjustments
             if (path.Alias.Equals(oldAlias))
             {
                var i = 0;
-               // change new alias and remember this for this formula to change formula strign right if nessesary
+               // change new alias and remember this for this formula to change formula string right if necessary
                while (formula.ObjectPaths.Select(x => x.Alias).Contains(newAlias))
                {
                   newAlias = $"{newAlias}{i}";
                   i++;
                }
 
-               _changes.Add(formula, _buildingBlock, new EditFormulaAliasCommand(formula, newAlias, oldAlias, _buildingBlock));
+               _changes.Add(formula, new EditFormulaAliasCommand(formula, newAlias, oldAlias, _buildingBlock));
             }
          }
 
          var explicitFormula = formula as ExplicitFormula;
-         if (explicitFormula == null) return;
+         if (explicitFormula == null)
+            return;
 
          // possible Formula string adjustments
          if (!containsWord(explicitFormula.FormulaString, oldAlias))
             return;
 
          var newFormulaString = wordReplace(explicitFormula.FormulaString, oldAlias, newAlias);
          var editFormulaStringCommand = new EditFormulaStringCommand(newFormulaString, explicitFormula, _buildingBlock);
-         _changes.Add(formula, _buildingBlock,
-            editFormulaStringCommand,
-            editFormulaStringCommand.Description);
+         _changes.Add(formula, editFormulaStringCommand, editFormulaStringCommand.Description);
       }
 
-      private string stringReplace(string replaceIn)
+      public void Visit(IObjectBase objectBase)
       {
-         return replaceIn.Replace(_oldName, _newName);
+         checkObjectBase(objectBase);
       }
 
-      private bool containsOldName(string stringToCheck)
+      public void Visit(NeighborhoodBuilder neighborhoodBuilder)
       {
-         return stringToCheck.Contains(_oldName);
+         checkObjectBase(neighborhoodBuilder);
+         //If not renaming a container or we are renaming a neighborhood builder, we exist
+         if (!(_objectToRename is IContainer) || !(_buildingBlock is SpatialStructure spatialStructure) || ReferenceEquals(neighborhoodBuilder, _objectToRename))
+            return;
+
+         //now the name of the container is used in the neighborhood name. Propose update
+         if (containsOldName(neighborhoodBuilder.Name))
+         {
+            var newNeighborhoodName = stringReplace(neighborhoodBuilder.Name);
+            _changes.Add(neighborhoodBuilder, renameCommand(neighborhoodBuilder, newNeighborhoodName), renameDescription(neighborhoodBuilder, neighborhoodBuilder.Name, newNeighborhoodName, _namePropertyName));
+         }
+
+         //check first and second neighbors
+         if (neighborhoodBuilder.FirstNeighborPath?.Contains(_oldName) ?? false)
```

Comment:
this code (174-178) is nearly identical to 180-184

Reply:
It can but I don't think this makes sense. There are two many differences in what is set or what is checked.

Reply:
makes sense